Update CCHeaderService after updates to CCCamera XML

menanteau opened the pull request.

menanteau committed:
Replace ccdNames by ccdLocations

menanteau committed:
Updates for new CCCamera ts_xml 5.1.0

menanteau committed:
Update to better handle indexed arrays of Telemetry

menanteau committed:
Updates to enable additionalKeys and additionalValues

Remove comments and update version

Formating and version update

menanteau committed:
Formatting and version update

menanteau committed:
Add DARKTIME. Move GroupID and imageType to addtionalKeys.

mareuter commented on bin/telemetry_sim_ComCam:
```diff
@@ -69,11 +69,16 @@ def cmdline():
+    parser.add_argument("--additionalKeys", action="store", default='imageType:groupId:testType',
+                        type=str, help="Optional additionalKeys")
+    parser.add_argument("--additionalValues", action="store", default='BIAS:CALSET_XXXX_YYYY:SIMTEST_ABCD',
+                        type=str, help="Optional additionalKeys")
```
Should be "Optional additionalValues"

menanteau replied:
Thanks for catching this.
Fixed.

mareuter commented on etc/ComCam/primary_hdu.header:
```diff
@@ -61,6 +61,7 @@ CONTRLLR=                      / The controller (e.g. O for OCS, C for (A)CCS)
 DAYOBS  =                      / The observation day as defined by image name
 SEQNUM  =                      / The sequence number from the image name
 GROUPID =                      /
+TESTTYPE=                      / Test Suite Type: DARK:FLAT:OBS:PPUMP:QE:SFLAT
```
BIAS can also be included here.

menanteau replied:
Fixed.

mareuter commented on python/HeaderService/hslib_salobj.py:
```diff
@@ -505,47 +508,54 @@ def extract_from_myData(self, keyword, myData, sep=":"):
-            self.log.debug(f"{keyword} is an array")
-            ccdnames = self.get_CCD_keywords(keyword, myData, sep)
-            self.log.info(f"For {keyword} extracted ccdnames: {ccdnames}")
+            self.log.debug(f"{keyword} is and array: CCD_array")
```
Should be "an array"

menanteau replied:
Thanks for catching my typos.

mareuter commented on python/HeaderService/hslib_salobj.py:
```diff
@@ -505,47 +508,54 @@ def extract_from_myData(self, keyword, myData, sep=":"):
             # Split the payload into an array of strings
             extracted_payload = dict(zip(ccdnames, payload.split(sep)))
+        elif self.config.telemetry[keyword]['array'] == 'indexed_array':
+            self.log.debug(f"{keyword} is and array: indexed_array")
```
Should be "an array"

menanteau replied:
Thanks for catching my typos.

mareuter commented on python/HeaderService/hslib_salobj.py:
```diff
@@ -505,47 +508,54 @@ def extract_from_myData(self, keyword, myData, sep=":"):
-            else:
-                # Otherwise take first element
-                extracted_payload = payload[0]
+            self.log.debug(f"{keyword} is just an array")
```
These three lines seem to imply a control block that looks to be missing. Please double check this.

menanteau replied:
It looks like that from the re-structuring of the function, but it's okay.

menanteau replied:
This is okay. It looks like something might be missing because of re structuring I did for the function.

mareuter commented on setup.cfg:
```diff
@@ -1,5 +1,5 @@
 max-line-length = 110
 max-doc-length = 79
-ignore = E133, E226, E228, N802, N803, N806, N812, N815, N816, W504
+ignore = E133, E226, E228, N802, N803, N806, N812, N815, N816, W504, F541, E741
```
You cleaned up a number of issues related to F541 so I don't see the need in ignoring it. E741 is a good thing to have and I don't see any instances in your code. As these are not on the curated list, any lines tripping these errors are supposed to be handled per this section **Lines that intentionally deviate from DM’s PEP 8 MUST include a noqa comment** found here: https://developer.lsst.io/python/style.html#code-may-be-validated-with-flake8.

menanteau replied:
Sorry, that was for testing only, and I forgot to put them back. I've done it now.

mareuter reviewed (changes requested):
Got a couple of things I'd like to see addressed before approving: possible missing control block and ignoring flake8 errors. The other things are minor.

menanteau committed:
Changes following code review.

Trying now with python7

Revert to python 3.6, fixing E741 travis error.

Fixing another F541

menanteau commented:
@mareuter I believe the changes are ready for you to review again.

menanteau committed:
Travis fix attempt

menanteau committed:
Reverting travis fix attempt.

mareuter reviewed (approved):
I think we are good now. PR approved.
